# MSc-Dissertation-2024

### Overview 

This project aims to investigate this effect using publicly available epigenomic, genomic, and transcriptomic datasets from cohorts of participants belonging into one of two clearly defined disease states; healthy or cancer. Testing how the different imputation methods will affect the supervised machine learning modelsâ€™ efficiency will give us insight into the most appropriate bioinformatic approach for future studies aiming at the detection of disease status or the prediction of disease risk based on molecular biomarkers.

### Methodology
* Search in public databases for epigenomic, genomic, and transcriptomic data from cohorts of two distinct disease states (healthy vs cancer)
* Preprocessing of raw data in order to attain all relevant features (methylation levels, SNPs, CNVs, gene expression levels) per individual
* Missing value imputation based on multiple methodologies (kNN, random forest, neural networks etc) Feature selection and machine learning predictor algorithms development
* Assessment of the optimal imputation method based on the performance of the machine learning predictor algorithms.

I have found 3 datasets: 
* WGBS
* RRBS
* EPIC microarray

## PRE-PROCESSING
### RRBS jobscripts explained

* 
